ludwig/schema/hyperopt/search_algorithm.py

```diff
@@ -1,22 +1,105 @@
 from dataclasses import field
-from typing import Dict
+from importlib import import_module
+from typing import Dict, List, Optional
 
 from marshmallow import fields, ValidationError
 
 from ludwig.api_annotations import DeveloperAPI
-from ludwig.hyperopt.registry import search_algorithm_registry  # Double-check this implicit import.
 from ludwig.schema import utils as schema_utils
+from ludwig.schema.hyperopt.utils import (
+    get_search_algorithm_dependencies,
+    get_search_algorithm_random_state_field,
+    register_search_algorithm,
+)
 from ludwig.schema.utils import ludwig_dataclass
 
 
+def metric_field(description: Optional[str] = None) -> fields.Field:
+    return schema_utils.String(
+        default=None,
+        allow_none=True,
+        description=description
+        or (
+            "Name of the metric used as objective in this experiment. This metric must be present in `raw_data` "
+            "argument to `log_data`. This metric must also be present in the dict reported/returned by the Trainable. "
+            "If `None` but a mode was passed, the `ray.tune.result.DEFAULT_METRIC` will be used per default."
+        ),
+    )
+
+
+def mode_field(description: Optional[str] = None) -> fields.Field:
+    return schema_utils.StringOptions(
+        options=["min", "max", None],
+        default=None,
+        allow_none=True,
+        description=description
+        or (
+            "One of `{min, max}`. Determines whether objective is minimizing or maximizing the metric attribute. "
```

Comment:
nit: Is None allowed? Is it meaningfully different from fallback to default (max)?

Reply:
`None` is the default in Ray Tune, but I'm actually considering dropping `metric` and `mode`from all of the search algs. We override those with `hyperopt.metric` and `hyperopt.goal` while setting up the search alg in `hyperopt/execution.py`, so I'm not sure it makes sense to expose those parameters in the search alg config.

Reply:
Followup: I dropped metric and mode as described above.